xmppserver/src/main/java/org/jivesoftware/openfire/net/SASLAuthentication.java

```diff
@@ -387,18 +389,15 @@ public static Status handle(LocalSession session, Element doc)
                         final LocalIncomingServerSession incomingServerSession = (LocalIncomingServerSession) session;
 
                         // Flag that indicates if certificates of the remote server should be validated.
-                        final boolean verify = JiveGlobals.getBooleanProperty( ConnectionSettings.Server.TLS_CERTIFICATE_VERIFY, true );
-                        if ( verify )
-                        {
-                            if ( verifyCertificates( incomingServerSession.getConnection().getPeerCertificates(), saslServer.getAuthorizationID(), true ) )
-                            {
-                                ( (LocalIncomingServerSession) session ).setAuthenticationMethod(ServerSession.AuthenticationMethod.SASL_EXTERNAL);
-                            }
-                            else
-                            {
-                                throw new SaslFailureException( Failure.NOT_AUTHORIZED, "Server-to-Server certificate verification failed." );
-                            }
+                        final boolean shouldVerifyCertificates = JiveGlobals.getBooleanProperty( ConnectionSettings.Server.TLS_CERTIFICATE_VERIFY, true );
```

Comment:
Does this guard against usage of a (non-self-signed) certificate that's signed by an entity that we do _not_ trust, when we allow for self-signed certificates? Maybe it's good to explicitly check if the certificate is self-signed?

Reply:
As above - we do not have the peers certificate for verification.